Description: The md5,sha modules are deprecated since Python 2.6.
 Replace them with hashlib.
Author: Alessio Treglia <alessio@debian.org>
Forwarded: https://bugzilla.gnome.org/show_bug.cgi?id=640387
---
 plugins/python/md5sum/md5sum.py   |   15 ++++++++-------
 plugins/python/sha1sum/sha1sum.py |   14 ++++++++------
 2 files changed, 16 insertions(+), 13 deletions(-)

--- gnome-commander.orig/plugins/python/md5sum/md5sum.py
+++ gnome-commander/plugins/python/md5sum/md5sum.py
@@ -20,19 +20,20 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
 
-import warnings
-
-warnings.simplefilter("ignore", DeprecationWarning)
-
 try:
     import gnomevfs
 except ImportError:
     import gnome.vfs as gnomevfs
 
 import os
+import sys
 import string
-import md5
-
+if sys.version_info[0] == 2 and sys.version_info[1] < 6:
+    import md5
+    md5_new = md5.new
+else:
+    from hashlib import md5
+    md5_new = md5
 
 def main(main_wnd_xid, active_cwd, inactive_cwd, selected_files):
     parent_dir = string.split(active_cwd, os.sep)[-1]
@@ -44,7 +45,7 @@ def main(main_wnd_xid, active_cwd, inact
             f = file(active_cwd+os.sep+uri.short_name, 'rb')
             file_content = f.read()
             f.close()
-            md5sum = md5.new(file_content).hexdigest()
+            md5sum = md5_new(file_content).hexdigest()
             f_md5sum.write('%s  %s\n' % (md5sum, uri.short_name))
     f_md5sum.close()
     return True
--- gnome-commander.orig/plugins/python/sha1sum/sha1sum.py
+++ gnome-commander/plugins/python/sha1sum/sha1sum.py
@@ -20,18 +20,20 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
 
-import warnings
-
-warnings.simplefilter("ignore", DeprecationWarning)
-
 try:
     import gnomevfs
 except ImportError:
     import gnome.vfs as gnomevfs
 
 import os
+import sys
 import string
-import sha
+if sys.version_info[0] == 2 and sys.version_info[1] < 6:
+    import sha
+    sha_new = sha.new
+else:
+    from hashlib import sha1
+    sha_new = sha1
 
 
 def main(main_wnd_xid, active_cwd, inactive_cwd, selected_files):
@@ -44,7 +46,7 @@ def main(main_wnd_xid, active_cwd, inact
             f = file(active_cwd+os.sep+uri.short_name, 'rb')
             file_content = f.read()
             f.close()
-            sha1sum = sha.new(file_content).hexdigest()
+            sha1sum = sha1_new(file_content).hexdigest()
             f_sha1sum.write('%s  %s\n' % (sha1sum, uri.short_name))
     f_sha1sum.close()
     return True
